### PR TITLE
add batch upload support

### PR DESCRIPTION
**Pull request by sadam21**

### CODE REVIEW

File: src/main/java/com/gooddata/dataset/DatasetException.java [hunk 1]
```diff
@@ -5,23 +5,36 @@
...
+    private final Collection<String> datasets;
 
     public DatasetException(String message, String dataset) {
         this(message, dataset, null);
     }
 
     public DatasetException(String message, String dataset, Throwable cause) {
-        super("Load dataset " + dataset + " failed: " + message, cause);
-        this.dataset = dataset;
+        super("Load dataset " + asList(dataset) + " failed: " + message, cause);
```
Comment (martiner) [inline src/main/java/com/gooddata/dataset/DatasetException.java]: rather call `this(message, asList(dataset), cause)`

File: src/main/java/com/gooddata/dataset/DatasetException.java [hunk 2]
```diff
@@ -5,23 +5,36 @@
...
+        this.datasets = datasets;
+    }
+
+    public DatasetException(String message, Collection<String> datasets) {
+        this(message, datasets, null);
     }
 
-    public String getDataset() {
-        return dataset;
+    public Collection<String> getDatasets() {
```
Comment (martiner) [inline src/main/java/com/gooddata/dataset/DatasetException.java]: preserve `public String getDataset()` to retain backward compatibility. eg. return `datasets.toString()`

File: src/main/java/com/gooddata/dataset/DatasetManifestList.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import org.codehaus.jackson.annotate.JsonProperty;
+
+import java.util.Collection;
+
+import static com.gooddata.util.Validate.notNull;
+
+/**
+ * Encapsulates list of {@link DatasetManifest}.
+ */
+public class DatasetManifestList {
```
Comment (martiner) [inline src/main/java/com/gooddata/dataset/DatasetManifestList.java]: does it need to be public? wouldnt `DatasetManifests` be a better name?

File: src/main/java/com/gooddata/dataset/DatasetService.java [hunk 4]
```diff
@@ -129,6 +107,99 @@ protected void onFinish() {
...
+            return pullLoad(project, dirPath, datasetsNames);
+        } catch (IOException e) {
+            throw new DatasetException("Unable to serialize manifest", datasetsNames, e);
+        } catch (DataStoreException | GoodDataRestException | RestClientException e) {
+            throw new DatasetException("Unable to load", datasetsNames, e);
+        }
+    }
+
+    private FutureResult<Void> pullLoad(Project project, final Path dirPath, final String dataset) {
+        List<String> datasets = new ArrayList<>();
```
Comment (martiner) [inline src/main/java/com/gooddata/dataset/DatasetService.java]: `asList()`?

File: src/main/java/com/gooddata/dataset/DatasetService.java [hunk 5]
```diff
@@ -129,6 +107,99 @@ protected void onFinish() {
...
+     * in case the ETL pull task fails
+     * @throws com.gooddata.dataset.DatasetException if there is a problem to serialize manifest or upload dataset
+     * @see <a href="https://developer.gooddata.com/article/multiload-of-csv-data">batch upload reference</a>
+     */
+    public FutureResult<Void> loadDatasets(final Project project, final Collection<DatasetManifest> datasets) {
+        notNull(project, "project");
+        notEmpty(datasets, "datasets");
+        for (DatasetManifest datasetManifest : datasets) {
+            if (datasetManifest.getSource() == null) {
+                throw new IllegalArgumentException(format("Source for dataset '%s' is null", datasetManifest.getDataSet()));
```
Comment (martiner) [inline src/main/java/com/gooddata/dataset/DatasetService.java]: check file and dataset as well

File: src/main/java/com/gooddata/dataset/DatasetException.java (hunk 1 above)
Comment (sadam21) [inline src/main/java/com/gooddata/dataset/DatasetException.java]: changed...

File: src/main/java/com/gooddata/dataset/DatasetException.java (hunk 2 above)
Comment (sadam21) [inline src/main/java/com/gooddata/dataset/DatasetException.java]: Method preserved.

File: src/main/java/com/gooddata/dataset/DatasetManifestList.java (hunk 3 above)
Comment (sadam21) [inline src/main/java/com/gooddata/dataset/DatasetManifestList.java]: Class made package protected.

File: src/main/java/com/gooddata/dataset/DatasetService.java (hunk 4 above)
Comment (sadam21) [inline src/main/java/com/gooddata/dataset/DatasetService.java]: Refactored.

File: src/main/java/com/gooddata/dataset/DatasetService.java (hunk 5 above)
Comment (sadam21) [inline src/main/java/com/gooddata/dataset/DatasetService.java]: Checks added...

**Pull request merged**